src/main/java/com/revature/assignforce/messaging/listeners/SkillListener.java

```diff
@@ -0,0 +1,61 @@
+package com.revature.assignforce.messaging.listeners;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.forwardedUrl;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.amqp.rabbit.annotation.Exchange;
+import org.springframework.amqp.rabbit.annotation.Queue;
+import org.springframework.amqp.rabbit.annotation.QueueBinding;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.amqp.support.AmqpHeaders;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.stereotype.Component;
+
+import com.rabbitmq.client.Channel;
+import com.revature.assignforce.beans.Focus;
+import com.revature.assignforce.beans.SkillIdHolder;
+import com.revature.assignforce.service.FocusService;
+
+@Component
+public class SkillListener {
+	
+	private FocusService focusService;
+	
+	private final String focusQueue;
+	
+	@RabbitListener(bindings = @QueueBinding(
+			value = @Queue(value = "focus-queue", durable = "true"),
+			exchange = @Exchange(value = "assignforce", ignoreDeclarationExceptions = "true"),
+			key = "assignforce.skills.delete")
+	)
+	
+	public void receiveMessage(final Integer skillId, Channel channel, 
+				@Header(AmqpHeaders.DELIVERY_TAG) long tag) {
+		try {
+			channel.basicAck(tag, false);
```

Comment:
The acknowledgement should not be sent until **after** the message has been processed. Otherwise there is no guarantee that the message will be correctly processed when it is permanently removed from the queue.